godot-core/src/lib.rs

```diff
@@ -77,14 +79,41 @@ pub mod private {
         F: FnOnce() -> R + std::panic::UnwindSafe,
         S: std::fmt::Display,
     {
-        match std::panic::catch_unwind(code) {
+        let location_line: Arc<Mutex<u32>> = Arc::new(Mutex::new(0));
+        let location_file: Arc<Mutex<String>> = Arc::new(Mutex::new(String::from("file")));
```

Comment:
I'd suggest making these `Arc<Mutex<Option<u32/String>>>` so we can detect when we were unable to get the line/file info.

Maybe you could even make this its own struct, something like (probably a better name than this)
```rs
struct GodotPanicInfo {
  line: u32,
  file: String,
}

// later

// can we use Default::default() here?
let info: Arc<Mutex<Option<GodotPanicInfo>>> = Arc::new(Mutex::new(None)); 
```